### DISCUSSION TITLE
Backups to BLOBs from SQL MIs using Managed Identity credentials

### DISCUSSION THREAD

**Discussion by alinselicean** [olahallengren/sql-server-maintenance-solution#780] [Questions & Answers]
Hello

We are using your tool to create a COPY_ONLY backup for our databases running on a number of SQL Managed Instances. However, due to some security requirements, we are not allowed to use SAS tokens, but instead use Managed Identity to control the access to URL blobs.

The DatabaseBackup SP is only checking for SAS credentials. I changed the code now to also include credentials using Managed Identity on my instances, but probably the best thing would be to just look if a credential with the required name exists.
![image](https://github.com/olahallengren/sql-server-maintenance-solution/assets/20295322/575c82a8-d6de-45ef-a0f1-b80d4ce4d2f5)

**Comment by FSalot** [comment]
Hello,
I'm facing the same issue with SQL Credential.
- It works in SSMS without using @Credential parameter
- It doesn't works when I'm using a powershell script even if I use @Credential parameter. I always have the error "The value for the parameter @Credential is not supported."

Do you have any solution for this issue?
have a nice day,
Florent

**Reply by FSalot** [reply]
Hello,
I'm not sure, you still have the issue...
We solved the issue by adding sysadmin right to the backup user. 

Without this right, the backup user seems to not be able to found the credential registred in SQL Instance.

Have a nice day,
Florent

**Comment by olahallengren** [comment]
I have been improving this in the 27 Oct 2024 version.
https://ola.hallengren.com/versions.html

Could you test it again?
